core/trino-main/src/test/java/io/trino/operator/output/TestPagePartitioner.java

```diff
@@ -272,126 +284,174 @@ public void testPartitionPositionsWithRleNotNull()
     {
-        PartitionedOutputOperator partitionedOutputOperator = partitionedOutputOperator(BIGINT).build();
+        PagePartitioner pagePartitioner = pagePartitioner(BIGINT).build();
         Page page = new Page(new DictionaryBlock(createLongsBlock(0), new int[] {0, 0, 0, 0}));
 
-        processPages(partitionedOutputOperator, page);
+        processPages(pagePartitioner, partitioningMode, page);
 
         List<Object> partition0 = readLongs(outputBuffer.getEnqueuedDeserialized(0), 0);
         assertThat(partition0).containsExactlyElementsOf(nCopies(4, 0L));
         List<Object> partition1 = readLongs(outputBuffer.getEnqueuedDeserialized(1), 0);
         assertThat(partition1).isEmpty();
     }
 
-    @Test
-    public void testOutputForViewDictionaryBlock()
+    @Test(dataProvider = "partitioningMode")
+    public void testOutputForViewDictionaryBlock(PartitioningMode partitioningMode)
     {
-        PartitionedOutputOperator partitionedOutputOperator = partitionedOutputOperator(BIGINT).build();
+        PagePartitioner pagePartitioner = pagePartitioner(BIGINT).build();
         Page page = new Page(new DictionaryBlock(createLongSequenceBlock(4, 8), new int[] {1, 0, 3, 2}));
 
-        processPages(partitionedOutputOperator, page);
+        processPages(pagePartitioner, partitioningMode, page);
 
         List<Object> partition0 = readLongs(outputBuffer.getEnqueuedDeserialized(0), 0);
         assertThat(partition0).containsExactlyInAnyOrder(4L, 6L);
         List<Object> partition1 = readLongs(outputBuffer.getEnqueuedDeserialized(1), 0);
         assertThat(partition1).containsExactlyInAnyOrder(5L, 7L);
     }
 
-    @Test(dataProvider = "types")
-    public void testOutputForSimplePageWithType(Type type)
+    @Test(dataProvider = "typesWithPartitioningMode")
+    public void testOutputForSimplePageWithType(Type type, PartitioningMode partitioningMode)
     {
-        PartitionedOutputOperator partitionedOutputOperator = partitionedOutputOperator(BIGINT, type).build();
+        PagePartitioner pagePartitioner = pagePartitioner(BIGINT, type).build();
         Page page = new Page(
                 createLongSequenceBlock(0, POSITIONS_PER_PAGE), // partition block
                 createBlockForType(type, POSITIONS_PER_PAGE));
         List<Object> expected = readChannel(Stream.of(page), 1, type);
 
-        processPages(partitionedOutputOperator, page);
+        processPages(pagePartitioner, partitioningMode, page);
 
         List<Object> partitioned = readChannel(outputBuffer.getEnqueuedDeserialized(), 1, type);
         assertThat(partitioned).containsExactlyInAnyOrderElementsOf(expected); // order is different due to 2 partitions joined
     }
 
+    @Test(dataProvider = "types")
+    public void testOutputWithMixedRowWiseAndColumnarPartitioning(Type type)
+    {
+        testOperatorOutputEqualsInput(type, PartitioningMode.COLUMNAR, PartitioningMode.ROW_WISE);
```

Comment:
`COLUMNAR, COLUMNAR` and `ROW_WISE, ROW_WISE` don't matter, do they?

Reply:
droped